synapse/config/server.py

```diff
@@ -228,6 +228,115 @@ def read_config(self, config):
             "show_users_in_user_directory", True,
         )
 
+        retention_config = config.get("retention")
+        if retention_config is None:
+            retention_config = {}
+
+        self.retention_enabled = retention_config.get("enabled", False)
+
+        retention_default_policy = retention_config.get("default_policy")
+
+        if retention_default_policy is not None:
+            self.retention_default_min_lifetime = retention_default_policy.get(
+                "min_lifetime"
+            )
+            if self.retention_default_min_lifetime is not None:
+                self.retention_default_min_lifetime = self.parse_duration(
```

Comment:
a thought: letting `parse_duration` take `None` (possibly protected by an `allow_none` param) would have saved a lot of boilerplate here!

Reply:
Agreed, but that's more something that should be done on mainline then ported to the dinsic branch rather than added to only the dinsic branch as part of this PR.